lua/nvim-treesitter/query.lua

```diff
@@ -61,9 +61,14 @@ local function runtime_queries(lang, query_name)
   return api.nvim_get_runtime_file(string.format('queries/%s/%s.scm', lang, query_name), true) or {}
 end
 
+local query_files_cache = {}
```

Comment:
Nice! We have a cache invalidation mechanism that is triggered when a user saves a queries/lang/xx.scm file. Can you have a look whether this could also apply here (e.g. when user creates a new query file It is found)

Reply:
@theHamsta done!

did some further testing and this is really a massive performance improvement.

Previously I had quite some lagging when editing neovim runtime files, and  all of that is completelty gone now!

Reply:
I'm pretty sure a lot of other issues related to slowness can probably be closed as well